packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -1177,11 +1188,7 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
 				continue;
 			}
 			const dataStoreId = pathParts[1];
-			assert(this.contexts.has(dataStoreId), 0x2d7 /* No data store with specified id */);
```

Comment:
This function is called in "test mode" which has slightly different semantics than prod. The idea being that in test mode, you can by pass the local data store assert and not need to create remote data stores. Let me look at the test though, it may be easy to fix.

Reply:
The only test that fails is "marks non-root data stores as referenced / unreferenced correctly" and it fails because extra error is found in the logs. Maybe I can "expect" this error and that's about it - not sure.

See here:
https://dev.azure.com/fluidframework/public/_build/results?buildId=245338&view=ms.vss-test-web.build-test-results-tab&runId=5357000&resultId=102206&paneView=debug

Reply:
It's in one of the GC tech debt to remove `updateUnusedRoutes` but we haven't got to it. You can add an expected log or skip these tests and assign a bug to me. I will fix them.

Reply:
Fixed it - https://github.com/microsoft/FluidFramework/pull/20087

Reply:
Nice to hear, as I was ready to give up :)
I can't use itExpects, as error happens only when running new runtime, and this test runs with all kinds of configs.
I can make itExpects error conditional on runtime version, but I do not see a way on provider to ask it - what version of runtime it uses.
The only other alterntive I see - expose EventAndErrorTrackingLogger.allowedErrors, which is more like "ignore if you see it" kind of system.

Reply:
My PR is approved and I will merge it soon. That might be the easiest way for you to move forward.

Reply:
Merged my PR.

Reply:
Merged your PR, tests are passing :)